compiler/rustc_typeck/src/impl_wf_check/min_specialization.rs

```diff
@@ -368,25 +386,57 @@ fn check_predicates<'tcx>(
     );
 
     for (predicate, span) in impl1_predicates {
-        if !impl2_predicates.contains(&predicate) {
+        if !impl2_predicates.iter().any(|pred2| trait_predicates_eq(predicate, *pred2)) {
```

Comment:
please add the following test here (that you can't have `~const Trait` in the default and `Trait` in the specializing impl)
```rust
#![feature(rustc_attrs, min_specialization, const_trait_impl)]

trait Trait {
    fn func() -> bool;
}

trait OtherTrait {}

impl<T> const Trait for T
where
    T: ~const OtherTrait,
{
    default fn func() -> bool {
        false
    }
}

#[rustc_specialization_trait]
trait Specialize {}

impl<T> const Trait for T
where
    T: OtherTrait, // missing ~const Bound
    T: Specialize,
{
    fn func() -> bool {
        true
    }
}
```
i would prefer to also check that in `trait_predicates_eq` even if the behavior is already like this.

after this r=me

Reply:
Won't this example fail to compile for other reasons? Doesn't the bound have to be `~const OtherTrait` in order for the impl to be well formed?

Reply:
why would it require `~const OtherTrait` for the impl to be well formed? Can a `const` impl not use non-const trait bounds?

Reply:
> Can a const impl not use non-const trait bounds?

I had assumed so, but that is apparently not correct. It only requires `~const` on the bound if you try to call any of the methods: https://play.rust-lang.org/?version=nightly&mode=debug&edition=2021&gist=4067fc6f900d74117a35567ee07dccb2

I'll fix this in the PR. When you say "and then r=me", do I need to do anything to make that happen after I update the PR?

Reply:
If you see "r=me" means once you fix the requested problem/nit you can then do "@bors r=lcnr" to approve this PR.

Reply:
yeah, in case you don't have merge rights, i can either delegate them for this pr, or approve this myself once that change is done